packages/cli/src/fulltext/querySymbols.ts

```diff
@@ -0,0 +1,78 @@
+import queryKeywords from './queryKeywords';
+import { readFileSync } from 'fs';
+
+export const SymbolRegexes: Record<string, RegExp[]> = {
+  cs: [
+    /(interface|class|enum|struct)\s+(?<symbol>\w+)/g,
+    /(\s|^)(?!using|try|catch|if|while|do|for|switch)(?<!#define\s+?)(?<symbol>[\w~$]+)\s*?\([^;)]*?\)[\w\s\d<>[\].:\n]*?{/g,
+  ],
+  cpp: [
+    /(struct|enum|union|class)\s+(?<symbol>\w+)\s*?\{/g,
+    /}\s*?(?<symbol>\w+)\s*?;/g,
+    /(\s|^)(?!try|catch|if|while|do|for|switch)(?<!#define\s+?)(?<symbol>[\w~$]+)\s*?\([^;)]*?\)[\w\s\d<>[\].:\n]*?{/g,
+  ],
+  rs: [/(struct|enum|union|trait|type|fn)\s+(?<symbol>[\w\p{L}]+)/gu],
+  go: [/type\s+(?<symbol>[\w\p{L}]+)/gu, /func\s+?(\(.*?\)\s*?)?(?<symbol>[\w\p{L}]+)\s*?\(/gu],
+  rb: [/(class|module)\s+(?<symbol>\w+)/g, /def\s+?(?<symbol>\w+)/g],
+  py: [/(class|def)\s+(?<symbol>\w+)/g],
+  java: [
+    /(class|@?interface|enum)\s+(?<symbol>[\w$]+)/g,
+    /(\s|^)(?!try|catch|if|while|do|for|switch)(?<symbol>[\w$]+)\s*?\([^;)]*?\)[\w\s\d<>[\].:\n]*?{/g,
+  ],
+  ts: [
+    /(class|interface|enum|type|function)\s+(?<symbol>[#$\w\p{L}]+)/gu,
+    /(\s|^)(?!using|try|catch|if|while|do|for|switch)(?<symbol>[#$\w\p{L}]+)\s*?\([^;)]*?\)[\w\s<>[\].:\n]*?\{/gu,
+    /(?<symbol>[#$\w\p{L}]+)\s*?(=|:)\s*?\(.*?\)\s*?=>/gu,
+  ],
+  kt: [/(class|typealias)\s+(?<symbol>[\w_]+)/g, /fun\s+?(<.+?>\s+)?(.*?\.)?(?<symbol>\w+)/g],
+  php: [/(class|trait|function)\s+(?<symbol>[\w\d_$]+)/g],
+};
+
+const genericSymbolRegexes: RegExp[] = [
+  /(^|\s)(?!using|try|catch|if|while|do|for|switch)(?<symbol>[#$\w\p{L}~]+)\s*?\(([^;)])*?\)[\w\s<>[\].:\n]*?\{/gmu,
+  /^(?!.*?(?:#|\/\/|"|')).*?(interface|class|enum|struct|union|trait|type(alias|def)?|fu?nc?(tion)?|module|def)\s+?(?<symbol>[#$\w\p{L}]+)/gmu,
+  /(?<symbol>[#$\w\p{L}~]+)\s*?=\s*?[\w\s<>[\].:\n]*?\{/gu,
+];
+
+// Define aliases for common file extensions
+['js', 'jsx', 'ts', 'tsx', 'vue', 'svelte'].forEach((ext) => {
+  SymbolRegexes[ext] = SymbolRegexes.ts;
+});
+
+['c', 'cc', 'cxx', 'h', 'hpp', 'cpp', 'hxx'].forEach((ext) => {
+  SymbolRegexes[ext] = SymbolRegexes.cpp;
+});
+
+function getUniqueMatches(source: string, regex: RegExp[]): string[] {
+  const matches = new Set(
+    regex
+      .flatMap((re) => Array.from(source.matchAll(re)))
+      .filter(Boolean)
+      .map(({ groups }) => groups?.symbol ?? '')
+  );
+  return Array.from(matches).sort();
```

Comment:
If a word occurs multiple times in a file, the general practice is to have the repeated occurrences of the word in the index. That will help to boost the file as a search match.

Reply:
Thanks, I've removed the `Set`.